lib/storage/src/content_manager/toc_components/toc_tmp_directories.rs

```diff
@@ -0,0 +1,195 @@
+    }
+
+    fn get_storage_temp_path(&self) -> PathBuf {
+        Path::new(self.storage_path()).join(TEMP_SUBDIR_NAME)
+    }
+
+    fn get_optional_temp_path(&self) -> Option<PathBuf> {
+        self.temp_path()
+            .map(|path| Path::new(path).join(TEMP_SUBDIR_NAME))
+    }
+
+    /// Get temporary storage path inside the `snapshots` directory.
+    pub fn snapshots_temp_path(&self) -> CollectionResult<PathBuf> {
+        let path = self.get_snapshots_temp_path();
+
+        if !path.exists() {
+            std::fs::create_dir_all(&path).map_err(|e| {
+                CollectionError::service_error(format!(
+                    "Failed to create snapshots temp directory at {}: {:?}",
+                    path.display(),
+                    e,
+                ))
+            })?;
+        }
+        Ok(path)
+    }
+
+    /// Get temporary storage path inside the `storage` directory.
+    pub fn storage_temp_path(&self) -> CollectionResult<PathBuf> {
+        let path = self.get_storage_temp_path();
+
+        if !path.exists() {
+            std::fs::create_dir_all(&path).map_err(|e| {
+                CollectionError::service_error(format!(
+                    "Failed to create storage temp directory at {}: {:?}",
+                    path.display(),
+                    e,
+                ))
+            })?;
+        }
+        Ok(path)
+    }
+
+    /// Get temporary storage path inside the `optional_temp_path` directory.
+    pub fn optional_temp_path(&self) -> CollectionResult<Option<PathBuf>> {
+        if let Some(path) = self.get_optional_temp_path() {
+            if !path.exists() {
+                std::fs::create_dir_all(&path).map_err(|e| {
+                    CollectionError::service_error(format!(
+                        "Failed to create optional temp directory at {}: {:?}",
+                        path.display(),
+                        e,
+                    ))
+                })?;
+            }
+            Ok(Some(path))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Get directory for snapshots-related temporary files.
+    /// If the optional_temp_path is specified, it will be used instead of snapshots_temp_path.
+    pub fn optional_temp_or_snapshot_temp_path(&self) -> CollectionResult<PathBuf> {
+        match self.optional_temp_path() {
+            Ok(Some(path)) => Ok(path),
+            Ok(None) => self.snapshots_temp_path(),
+            Err(err) => Err(err),
+        }
+    }
+
+    /// Get directory for storage-related temporary files.
+    /// If the optional_temp_path is specified, it will be used instead of storage_temp_path.
+    pub fn optional_temp_or_storage_temp_path(&self) -> CollectionResult<PathBuf> {
+        match self.optional_temp_path() {
+            Ok(Some(path)) => Ok(path),
+            Ok(None) => self.storage_temp_path(),
+            Err(err) => Err(err),
+        }
+    }
+
+    pub fn upload_dir(&self) -> CollectionResult<PathBuf> {
```

Comment:
Is someone responsible for cleaning the `upload` folder?

Reply:
It is a subdir for `tmp` which is autocleaned on restart

Reply:
And during the run actix also cleans it after uploading